bayes_opt/constraint.py

```diff
@@ -48,13 +53,7 @@ def __init__(self, fun, lb, ub, random_state=None):
             msg = "Lower bounds must be less than upper bounds."
             raise ValueError(msg)
 
-        basis = lambda: GaussianProcessRegressor(
-            kernel=Matern(nu=2.5),
-            alpha=1e-6,
-            normalize_y=True,
-            n_restarts_optimizer=5,
-            random_state=random_state,
-        )
+        basis = partial(_create_gaussian_process_regressor, random_state=random_state)
```

Comment:
Is this considered a better style compared to using `lambda`? It seems that this obscures to some degree what's happening.

Reply:
The use of `lambda` in `python` is not recommended except in some case. 
I have modified the code above to add the `ruff` rule accordingly.

If this code is obscure, there are two ways to go about it.
1. ignore the `ruff` rule in your existing code
2. change `lambda` to `def`